src/rules/tictactoepoker/tictactoepoker.ts

```diff
@@ -0,0 +1,752 @@
+                    return false;
+                }
+
+                return true;
+            })();
+
+
+            // Straight (just three consecutive cards)
+            const is_straight: boolean = (() => {
+                if (cards[0].get_value() + 1 == cards[1].get_value() &&
+                    cards[1].get_value() + 1 == cards[2].get_value()) {
+                        return true;
+                }
+            })();
+
+            if (is_straight && is_flush) {
+                return {combination_type: CombinationType.StraightFlush, rank_card: cards[2]};
+            }
+            if (is_flush) {
+                // Find the card with the highest rank value - it can be anywhere in the three.
+                let best_rank_value = 0;
+                let best_card = Card.EMPTY;
+                for (let card of cards) {
+                    if (card.get_rank_value() > best_rank_value) {
+                        best_rank_value = card.get_rank_value();
+                        best_card = card;
+                    }
+                }
+                return {combination_type: CombinationType.Flush, rank_card: best_card};
+            }
+            if (is_straight) {
+                return {combination_type: CombinationType.Straight, rank_card: cards[2]};
+            }
+
+            // Three of a kind
+            if (cards[0].get_value() == cards[1].get_value() &&
+                cards[1].get_value() == cards[2].get_value()) {
+                return {combination_type: CombinationType.ThreeOfAKind, rank_card: cards[0]};
+            }
+        }
+
+        // Pair
+        const pair_card: Card = (() => {
+            for (let i = 0; i < cards.length; i++) {
+                for (let j = i + 1; j < cards.length; j++) {
+                    if (cards[i].get_value() == cards[j].get_value()) {
+                        return cards[i];
+                    }
+                }
+            }
+            return Card.EMPTY;
+        })();
+
+        if (pair_card != Card.EMPTY) {
+            return {combination_type: CombinationType.Pair, rank_card: pair_card};
+        }
+
+        return {combination_type: CombinationType.None};
+    });
+
+    const is_better = (score_a, score_b) => {
+        if (score_a.combination_type == score_b.combination_type) {
+            if (score_a.combination_type == CombinationType.None) {
+                return true;
+            }
+            return score_a.rank_card.get_rank_value() >= score_b.rank_card.get_rank_value();
+        } else {
+            return score_a.combination_type > score_b.combination_type;
+        }
+    };
+
+    // Since Aces can be 1 or 14, we have to try all combinations.
+    //
+    // Note the same Ace on the board can be treated differently
+    // for the different directions (rows, columns, diagonals)
+    let score = {combination_type: CombinationType.None};
+    let ace_state = [1, 1, 1];
+    for (ace_state[0] of [1, 14]) {
```

Comment:
this enumerates all ace possibilities, even if there are less than 3 aces in the hand?

Reply:
Updated to only enumerate relevant possibilities.